cfn-resources/ldap-configuration/cmd/resource/resource.go

```diff
@@ -0,0 +1,295 @@
+}
+
+func Get(client *mongodbatlas.Client, groupID string) (*mongodbatlas.LDAPConfiguration, *handler.ProgressEvent) {
+	ldapConf, res, err := client.LDAPConfigurations.Get(context.Background(), groupID)
+	if err != nil {
+		errPe := progressEvents.GetFailedEventByResponse(err.Error(), res.Response)
+		return nil, &errPe
+	}
+
+	if !isResourceEnabled(*ldapConf) {
+		errPe := progressEvents.GetFailedEventByCode("Authentication is disabled for the selected project", cloudformation.HandlerErrorCodeNotFound)
+		return nil, &errPe
+	}
+
+	return ldapConf, nil
+}
+
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	// Validation
+	modelValidation := validator.ValidateModel(UpdateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Debugf("Update - error: %+v", err)
+		return progressEvents.GetFailedEventByCode(err.Error(), cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	// Validate if resource exists
+	_, errPe := Get(client, *currentModel.GroupId)
+	if errPe != nil {
+		return *errPe, nil
+	}
+
+	ldapReq := currentModel.GetAtlasModel()
+
+	LDAPConfigResponse, res, err := client.LDAPConfigurations.Save(context.Background(), *currentModel.GroupId, ldapReq)
+	if err != nil {
+		_, _ = log.Debugf("Create - error: %+v", err)
+		return progressEvents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	currentModel.CompleteByResponse(*LDAPConfigResponse)
+
+	if err != nil {
+		_, _ = log.Debugf("Update - error: %+v", err)
+		return progressEvents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	// Response
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	// Validation
+	modelValidation := validator.ValidateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Debugf("Delete - error: %+v", err)
+		return progressEvents.GetFailedEventByCode(err.Error(), cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	// Validate if resource exists
+	_, errPe := Get(client, *currentModel.GroupId)
+	if errPe != nil {
+		return *errPe, nil
+	}
+
+	return executeManualDelete(currentModel), nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	return handler.ProgressEvent{}, errors.New("not implemented: List")
+}
+
+func executeManualDelete(currentModel *Model) handler.ProgressEvent {
```

Comment:
[q] Why do you need this? isn't calling the delete endpoint enough?

Reply:
is explained on the note, the ldap configurations "delete" is not deleting the resource. is only reomving one user from the userToDNMapping, as explaned in the https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/LDAP-Configuration/operation/removeOneCurrentLdapUserToDnMapping, and the aws resource is expecting the resource to be disabled. so we had to execute the "Save" function with the authenticationEnabled": false,"authorizationEnabled" : false, but the problem is that when executing this with the client method, is not taking those parameters as false: 
![image](https://user-images.githubusercontent.com/69170650/209832556-e54eb684-ee97-4f40-8cd3-069067c08445.png)